l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,249 @@
+    mapping(uint256 => address) public stateTransitionManager;
+    /// @notice chainID => base token address
+    mapping(uint256 => address) public baseToken;
+    /// @notice chainID => bridge holding the base token
+    /// @notice a bridge can have multiple tokens
+    mapping(uint256 => address) public baseTokenBridge;
+
+    IL1Bridge public wethBridge;
+
+    modifier onlystateTransitionManager(uint256 _chainId) {
+        require(msg.sender == stateTransitionManager[_chainId], "Bridgehub: not state transition");
+        _;
+    }
+
+    modifier onlystateTransitionManagerChain(uint256 _chainId) {
+        require(
+            msg.sender == IStateTransitionManager(stateTransitionManager[_chainId]).stateTransition(_chainId),
+            "Bridgehub: not state transition chain"
+        );
+        _;
+    }
+
+    modifier onlyBaseTokenBridge(uint256 _chainId) {
+        require(msg.sender == baseTokenBridge[_chainId], "Bridgehub: not base token bridge");
+        _;
+    }
+
+    function initialize() external reentrancyGuardInitializer returns (bytes32) {
+    }
+
+    ///// Getters
+
+    function getZkSyncStateTransition(uint256 _chainId) public view returns (address) {
+        return IStateTransitionManager(stateTransitionManager[_chainId]).stateTransition(_chainId);
+    }
+
+    //// Registry
+
+    /// @notice State Transition can be any contract with the appropriate interface/functionality
+    function newstateTransitionManager(address _stateTransitionManager) external onlyOwner {
+        require(!stateTransitionManagerIsRegistered[_stateTransitionManager], "Bridgehub: state transition already registered");
+        stateTransitionManagerIsRegistered[_stateTransitionManager] = true;
+    }
+
+    /// @notice State Transition can be any contract with the appropriate interface/functionality
+    /// @notice this stops new Chains from using the STF, old chains are not affected
+    function removestateTransitionManager(address _stateTransitionManager) external onlyOwner {
+        require(stateTransitionManagerIsRegistered[_stateTransitionManager], "Bridgehub: state transition already registered");
+        stateTransitionManagerIsRegistered[_stateTransitionManager] = false;
+    }
+
+    /// @notice token can be any contract with the appropriate interface/functionality
+    function newToken(address _token) external onlyOwner {
+        require(!tokenIsRegistered[_token], "Bridgehub: token already registered");
+        tokenIsRegistered[_token] = true;
+    }
+
+    /// @notice Bridge can be any contract with the appropriate interface/functionality
+    function newTokenBridge(address _tokenBridge) external onlyOwner {
+        require(!tokenBridgeIsRegistered[_tokenBridge], "Bridgehub: token bridge already registered");
+        tokenBridgeIsRegistered[_tokenBridge] = true;
+    }
+
+    function setWethBridge(address _wethBridge) external onlyOwner {
+        wethBridge = IL1Bridge(_wethBridge);
```

Comment:
It is never used, why?

Reply:
it should be used, we shoul only allow a single bridge for eth deposits